src/core/agent/records/createOrUpdateBasicRecord.ts

```diff
@@ -0,0 +1,11 @@
+import { BasicRecord } from "./basicRecord";
+import { Agent } from "../agent";
+
+export async function createOrUpdateBasicRecord(record: BasicRecord) {
```

Comment:
I would move this function to `basicStorage.ts` - it can just be at the bottom of the file outside the class. No need for a whole file for this

Probably makes the test harder to mock but worth it for maintainability